beacon-chain/rpc/prysm/v1alpha1/validator/proposer_attestations.go

```diff
@@ -309,3 +317,211 @@ func (vs *Server) deleteAttsInPool(ctx context.Context, atts []ethpb.Att) error
 	}
 	return nil
 }
+
+// isAttestationFromCurrentEpoch returns true if the attestation is from the current epoch.
+func (vs *Server) isAttestationFromCurrentEpoch(att ethpb.Att) bool {
+	currentSlot := vs.TimeFetcher.CurrentSlot()
+	currentEpoch := slots.ToEpoch(currentSlot)
+	attestationSlot := att.GetData().Slot
+	attestationEpoch := slots.ToEpoch(attestationSlot)
+	return attestationEpoch == currentEpoch
+}
+
+// isAttestationFromPreviousEpoch returns true if the attestation is from the previous epoch.
+func (vs *Server) isAttestationFromPreviousEpoch(att ethpb.Att) bool {
+	currentSlot := vs.TimeFetcher.CurrentSlot()
+	previousEpoch := slots.ToEpoch(currentSlot)
+	if previousEpoch >= 1 {
+		previousEpoch = previousEpoch.Sub(1)
+	}
```

Comment:
Instead of a sub here it's better to return `attestationEpoch + 1 == currentEpoch` in the return. No need to worry about an overflow either because we never process attestations from the future.